src/skmatter/neighbors/_sparsekde.py

```diff
@@ -0,0 +1,601 @@
+
+    Parameters
+    ----------
+    descriptors: np.ndarray
+        Descriptors of the system where you want to build a sparse KDE.
+        It should be an array of shape `(n_descriptors, n_features)`.
+    weights: np.ndarray, default=None
+        Weights of the descriptors.
+        If None, all weights are set to `1/n_descriptors`.
+    kernel : {'gaussian'}, default='gaussian'
+        The kernel to use. Currentlty only one.
+    metric : str, default='periodic_euclidean'
+        The metric to use. Currently only one.
+    metric_params : dict, default=None
+        Additional parameters to be passed to the use of
+        metric.  i.e. the cell dimension for `periodic_euclidean`
+        {'cell': [2, 2]}
+    fspread : float, default=-1.0
+        The fractional "space" occupied by the voronoi cell of each grid. Use this when
+        each cell is of a similar size.
+    fpoints : float, default=0.15
+        The fractional number of points in the voronoi cell of each grid points. Use
+        this when each cell has a similar number of points.
+
+
+    Attributes
+    ----------
+    kdecut2 : float
+        The cut-off value for the KDE.
+    cell : numpy.ndarray
+        The cell dimension for the metric.
+    model : :class:`skmatter.utils.GaussianMixtureModel`
+        The model of the KDE.
+    cluster_mean : numpy.ndarray of shape (n_clusters, n_features)
+        The mean of each gaussian.
+    cluster_cov : numpy.ndarray of shape (n_clusters, n_features)
+        The covariance matrix of each gaussian.
+    cluster_weight : numpy.ndarray of shape (n_clusters, n_features)
+        The weight of each gaussian.
+
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from skmatter.neighbors import SparseKDE
+    >>> from skmatter.feature_selection import FPS
+    >>> np.random.seed(0)
+    >>> n_samples = 10000
+    >>> cov1 = [[1, 0.5], [0.5, 1]]
+    >>> cov2 = [[1, 0.5], [0.5, 0.5]]
+    >>> sample1 = np.random.multivariate_normal([0, 0], cov1, n_samples)
+    >>> sample2 = np.random.multivariate_normal([4, 4], cov2, n_samples)
+    >>> samples = np.concatenate([sample1, sample2])
+    >>> selector = FPS(n_to_select=int(np.sqrt(2 * n_samples)))
+    >>> result = selector.fit_transform(samples.T).T
+    >>> estimator = SparseKDE(samples, None, fpoints=0.5)
+    >>> estimator.fit(result)
+    SparseKDE(descriptors=array([[-1.72779275, -1.32763554],
+           [-1.96805856,  0.27283464],
+           [-1.12871372, -2.1059916 ],
+           ...,
+           [ 3.75859454,  3.10217702],
+           [ 1.6544348 ,  3.41851374],
+           [ 4.08667637,  3.42457743]]),
+              fpoints=0.5,
+              weights=array([5.e-05, 5.e-05, 5.e-05, ..., 5.e-05, 5.e-05, 5.e-05]))
+    >>> round(estimator.score(result), 3)
+    -759.831
+    """
+
+    def __init__(
+        self,
+        descriptors: np.ndarray,
+        weights: Union[np.ndarray, None] = None,
+        kernel: str = "gaussian",
```

Comment:
kernel could also be an executable instead of the string. Or is this the scikit-learn way to pass strings and get the functions from a dictionary. Also asking @agoscinski here.

Reply:
This parameter is actually invalid now, because the gaussian kernel is hardcoded in this class. Perhaps it should be deleted, since currently I do not see a way to decouple the gaussian kernel from this class and adapt other kernels.

Reply:
Okay, then we can remove it and mention in the docs the we only support Gaussian kernels!